test/installers/knative_installer.go

```diff
@@ -0,0 +1,138 @@
+		return framework.WaitForPodsInNamespace(knativeEventingNamespace, knativeEventingNumberOfPods, 3)
+	}
+
+	framework.GetMainLogger().Info("Installing Knative eventing")
+
+	output, err := framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/eventing-crds.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing CRDs failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing CRDs")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/eventing-core.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing core components failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing core components")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/in-memory-channel.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing in-memory channel failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing in-memory channel")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/mt-channel-broker.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing mt-channel broker failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing mt-channel broker")
+	}
+
+	// Start namespace logging
+	go func() {
+		err := framework.StartPodLogCollector(knativeEventingNamespace)
+		if err != nil {
+			framework.GetLogger(knativeEventingNamespace).Error(err, "Error starting log collector", "namespace", knativeEventingNamespace)
+		}
+	}()
+
+	return framework.WaitForPodsInNamespace(knativeEventingNamespace, knativeEventingNumberOfPods, 3)
+}
+
+func uninstallKnativeEventingUsingYaml() error {
+	framework.GetMainLogger().Info("Uninstalling Knative eventing")
+
+	// Stop namespace logging
+	if err := framework.StopPodLogCollector(knativeEventingNamespace); err != nil {
+		framework.GetMainLogger().Error(err, "Error stopping log collector", "namespace", knativeEventingNamespace)
+	}
+
+	output, err := framework.CreateCommand("oc", "delete", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/mt-channel-broker.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Deleting eventing mt-channel broker failed, output: %s", output))
+		return fmt.Errorf("Error deleting eventing mt-channel broker")
```

Comment:
Returning means we don't even try to remove other resources.
Shouldn't we log and then at the end of all deletion return error if any happens ?

Reply:
+1